packages/@ourworldindata/grapher/src/slopeCharts/SlopeChart.tsx

```diff
@@ -324,6 +326,77 @@ export class SlopeChart
         return legendBins.some((bin) => colorsInUse.includes(bin.color))
     }
 
+    @computed
+    private get selectedEntitiesWithoutData(): string[] {
+        const { transformedTable, selectedEntityNames } = this
+
+        const table = transformedTable.filterByEntityNames(selectedEntityNames)
+
+        // group by entity name
+        const tableByEntityName = keyBy<OwidTable>(
+            table.groupBy(table.entityNameColumn.slug),
+            (table) => table.availableEntityNames[0]
+        )
+
+        return this.selectedEntityNames.filter((entityName) => {
+            const table = tableByEntityName[entityName]
+            const yColumn = table.get(this.yColumnSlug)
+            return yColumn.numValues < 2
+        })
+    }
+
+    @computed private get noDataSection(): JSX.Element {
+        const { selectedEntitiesWithoutData: noDataEntities } = this
+
+        const nDisplayedEntities = 5
+        const displayedEntities = noDataEntities.slice(0, nDisplayedEntities)
+        const nRemainingEntities = Math.max(
+            0,
+            noDataEntities.length - nDisplayedEntities
+        )
+
+        const bounds = new Bounds(
+            this.legendX,
+            this.legendY + this.legendHeight,
+            this.sidebarWidth,
+            this.bounds.height - this.legendHeight
+        )
+
+        return (
+            <foreignObject
```

Comment:
Using `foreignObject` here so that we get line breaks for free. Our static charts can't render foreign objects in SVG, but I don't think there is a need for static charts to include the No data list. It's added as additional feedback if a user selects an entity that has no data.

Reply:
Yes, makes sense to me!